cpmpy/transformations/flatten_model.py

```diff
@@ -621,3 +633,46 @@ def negated_normal(expr):
             return negated_normal(Operator('and', expr.decompose()))
         else:
             raise NotImplementedError("negate_normal {}".format(expr))
+
+
+def _wsum_should_flatten(arg):
```

Comment:
Check these additions with pull request #182 and with @tias  :)

Reply:
that commit was about being conservative at expression creation time about when to create a wsum, e.g. the expression should be close to what the user wrote.

Here we are in flatten, and it makes sense to do some additional optimisations to avoid unnecessary auxiliary variable creation (which is what this is).

Could it make sense/be possible to separate the addition of these optimisations from the wsum branch? Yes, less expressions will be supported then in this branch (because any auxiliary intvar created is not supported), but then I can look at the optimisation at a different pace and with e.g. a list of example expressions we want to optimize...